openedx/core/djangoapps/discussions/handlers.py

```diff
@@ -66,7 +66,10 @@ def update_course_discussion_config(configuration: CourseDiscussionConfiguration
                 topic_link.enabled_in_context = False
             else:
                 topic_link.enabled_in_context = True
+                topic_link.ordering = topic_context.ordering
                 topic_link.title = topic_context.title
+                if topic_context.external_id:
+                    topic_link.external_id = topic_context.external_id
```

Comment:
Save the ordering to the database, this allows efficient sorting on the server side without needing to load up the whole course structure. 

Alternatively, we will need to load the course structure in the frontend, so we can do the sorting there.